src/components/Navigation/NavBar/NavItems/NavItem/NavItem.js

```diff
@@ -1,10 +1,16 @@
 import React from 'react';
 import classes from './NavItem.module.css'
+import PropTypes from 'prop-types'
 
 const navItem = (props) => (
```

Comment:
Rather than accessing props as `props.whatever`, you can destructure the `props` argument directly in the function signature, like
```
const navItem = ({clicked, children, ariaLable}) => (
``` 

Handy because it a) declutters all the `props.whatever`'s, and b) means you can glance at the function signature to see all the props, rather than search through function body.

Can do the same if one of the arguments is, an array, e.g. `([left, right]) =>`

If the array has arbitrary length, but you care in particular about a fixed number of left values, you can use array "rest" like this (note: "rest" and "spread" operators share the `...` syntax, which is which is up to context)
```
([left, right, ...the_rest]) =>
```
in which case `the_rest` is either an empty array or an array of all the remaining elements. Object spread/rest works similarly, but is key name based rather than position based.

Can also do the same with variable declaration, e.g.
```
const [first_entry, second_entry, ...array_of_other_entries] = some_array;
const {named_key, other_named_key, ...object_with_other_values} = some_object;
```

Reply:
I've seen the usage of destructured props in typescript but never gave much thought of using it in js! Good to know it works the same way. I'll slowly implement this method across the application. I didn't know about the rest operator before. I put some research into it and it looks like it could be useful!